Adding SDK for litmus-python and updating base code changes (#19)
litmuschaos/litmus-python: contribute/developer-guide/README.md

## Steps to Bootstrap a Chaos Experiment

The artifacts associated with a chaos-experiment are summarized below: 

- Submitted in the litmuschaos/litmus-python repository, under the experiments/*chaoscategory*/*experiment-name* folder 

>   - Experiment business logic in golang. May involve creating new or reusing an existing chaosLib 

Reviewer: ```suggestion
  - Experiment business logic in python. May involve creating new or reusing an existing chaosLib 
```

  - Experiment test deployment manifest that is used for verification purposes
  - Experiment RBAC (holds experiment-specific ServiceAccount, Role and RoleBinding)

>   Example: [pod delete experiment in litmus-python](/experiments/generic/pod-delete)

Reviewer: will it redirect to the corresponding experiment?

  - Experiment RBAC (holds experiment-specific ServiceAccount, Role and RoleBinding)
  - Experiment Engine (holds experiment-specific chaosengine)

  Example: [pod delete experiment in chaos-charts](https://github.com/litmuschaos/chaos-charts/tree/master/charts/generic/pod-delete)

The *generate_experiment.py* script is a simple way to bootstrap your experiment, and helps create the aforementioned artifacts in the 
appropriate directory (i.e., as per the chaoscategory) based on an attributes file provided as input by the chart-developer. The 
scaffolded files consist of placeholders which can then be filled as desired.  

### Pre-Requisites

- *python3* is available (`sudo apt-get install python3.6`) 
- *jinja2* & *pyYaml* python packages are available (`sudo apt-get install python3-pip`, `pip install jinja2`, `pip install pyYaml`) 

### Steps to Generate Experiment Manifests

- Clone the litmus-python repository & navigate to the `contribute/developer-guide` folder

  ```
  $ git clone https://github.com/litmuschaos/litmus-python.git
  $ cd litmus/contribute/developer-guide
  ```

> - Populate the `attributes.yaml` with details of the chaos experiment (or chart). Use the [attributes.yaml.sample](/contribute/developer-guide/attributes.yaml.sample) as reference. 

Reviewer: provide the correct url?

  keywords:
    - "pods"
    - "kubernetes"
    - "samplecategory"
    - "exec"
  platforms:
    - Minikube
  scope: "Namespaced"
  auxiliaryappcheck: false
  permissions:
    - apigroups:
        - ""
        - "batch"
        - "apps"
        - "litmuschaos.io"
      resources:
        - "jobs"
        - "pods"
        - "pods/log"
        - "events"
        - "deployments"
        - "replicasets"
        - "pods/exec"
        - "chaosengines"
        - "chaosexperiments"
        - "chaosresults"
      verbs:
        - "create"
        - "list"
        - "get"
        - "patch"
        - "update"
        - "delete"
        - "deletecollection"
  maturity: "alpha"
  maintainers:
    - name: "oumkale"
      email: "oumkale@chaosnative.com" 
  provider:
    name: "ChaosNative"
  minkubernetesversion: "1.12.0"
  references:
    - name: Documentation
      url: "https://docs.litmuschaos.io/docs/getstarted/"

  ```

- Run the following command to generate the necessary artefacts for submitting the `samplecategory` chaos chart with 
  `pod-delete` experiment.

  ```

>   $ python3 generate_experiment.py --attributes_file=attributes.yaml --generate_type=experiment

Reviewer: Can we keep the same flag names as of litmus-go repo?

  ```

>   **Note**: Replace the `generate_type=<experiment>` placeholder with the appropriate value based on the usecase: 

Reviewer: ```suggestion
  **Note**: Provide the `generate_type` flag based on the usecase: 
```

  keywords:
    - "pods"
    - "kubernetes"
    - "sample-category"
    - "exec"
  platforms:
    - Minikube
  scope: "Namespaced"
  auxiliaryappcheck: false
  permissions:
    - apigroups:
        - ""
        - "batch"
        - "apps"
        - "litmuschaos.io"
      resources:
        - "jobs"
        - "pods"
        - "pods/log"
        - "events"
        - "deployments"
        - "replicasets"
        - "pods/exec"
        - "chaosengines"
        - "chaosexperiments"
        - "chaosresults"
      verbs:
        - "create"
        - "list"
        - "get"
        - "patch"
        - "update"
        - "delete"
        - "deletecollection"
  maturity: "alpha"
  maintainers:
    - name: "oumkale"
      email: "oumkale@chaosnative.com" 
  provider:
    name: "ChaosNative"
  minkubernetesversion: "1.12.0"
  references:
    - name: Documentation
      url: "https://docs.litmuschaos.io/docs/getstarted/"

  ```

- Run the following command to generate the necessary artefacts for submitting the `sample_category` chaos chart with 
  `sample_exec_chaos` experiment.

  ```

>   $ python3 generate_experiment.py -f=attributes.yaml -g=<experiment>

Reviewer: ```suggestion
  $ python3 generate_experiment.py -f=attributes.yaml -g=experiment
```
